crates/sui/tests/ptb_files_tests.rs

```diff
@@ -63,30 +122,48 @@ async fn test_build_and_run_ptb_from_files(path: &Path) -> datatest_stable::Resu
     Ok(())
 }
 
-async fn test_no_gas_picker_from_files(path: &Path) -> datatest_stable::Result<()> {
+#[cfg_attr(not(msim), tokio::main)]
+async fn test_no_gas_picker_from_files(path: &Path) -> Result<(), Box<dyn std::error::Error>> {
     let ptb = PTB::default();
     let cmd = PTB::command();
     let file = path.to_str().unwrap();
     let args = cmd.get_matches_from(vec!["ptb", "--file", file]);
+    let cwd = std::env::current_dir().unwrap();
+    let commands = ptb.from_matches(cwd, &args, &mut BTreeMap::new()).unwrap();
+    let mut parser = PTBParser::new();
+    for (_, cmd) in commands.clone() {
+        parser.parse(cmd.clone());
+    }
+    let (parsed, _errors) = parser.finish();
+
     let test_cluster = TestClusterBuilder::new().build().await;
     let context = test_cluster.wallet;
+    let client = context.get_client().await?;
+
+    let built_ptb = ptb.parse_and_build_ptb(parsed, &context, client).await;
+    assert!(built_ptb.is_err());
+
+    let rendered = render_errors(commands, built_ptb.err().unwrap());
+    let mut results = vec![];
+    for e in rendered.iter() {
+        results.push(format!("{}", e));
+    }
 
-    let execute = ptb.execute(args, Some(context)).await;
-    assert!(execute.is_err());
+    insta::assert_display_snapshot!(format!("error_gas_picker"), results.join("\n"));
     Ok(())
 }
 
 datatest_stable::harness!(
+    test_ptb_preview,
+    TEST_DIR,
+    r".*\.ptb$",
     test_parse_all_ptb_files,
-    "tests/ptb_files",
-    r"^*.ptb$",
+    TEST_DIR,
+    r".*\.ptb$",
     test_parse_bad_ptb_files,
-    "tests/ptb_files",
-    r"^*.bad",
-    // test_build_and_run_ptb_from_files,
-    // "tests/ptb_files",
-    // r"^run_*.sh",
-    // test_no_gas_picker_from_files
-    // "tests/ptb_files",
-    // "no_gas_picker",
+    TEST_DIR,
+    r"^.*\.bad$",
```

Comment:
I'm not sure if we need all of these separate test runners, since we can allow on the expected value files to check for errors/lack of errors for tests and to snapshot the test at different parts of the flow I think. 

What are your thoughts on having a single test runner (e.g., `test_ptb_file`) that will
* Save the preview to the snapshot if the preview flag is present
* Parse the PTB, and save the parsed PTB (/any errors) to the same snapshot
* Build the PTB if no parsing errors, and save the built PTB (/any errors) to the same snapshot.

My thinking is this should then lead to a pretty uniform/easy way of writing tests for this, and have a good (bijective) mapping of test -> snapshot file.

Reply:
We can do that. So we'd have a few PTB files that are OK and go through from start to end, including execution, and a few that fail in different parts?